scripts/seastar-json2code.py

```diff
@@ -256,68 +257,120 @@ def resolve_model_order(data):
             models.add(model_name)
     return res
 
+
+def not_first():
+    '''
+    Returns True when gets called for the first time, False otherwise
+
+    used as the predicate parameter of textwrap.indent(), so the first
+    line is not indented. this helps to preserve the Python code's logical
+    indention in the template, and allows us to put something like::
+
+      blah = textwrap.dedent("""\
+           foo bar
+               blah blah
+           foo bar
+    """
+    '''
+    _is_first = True
+
+    def should_indent(_):
+        nonlocal _is_first
+        first = _is_first
+        _is_first = False
+        return not first
+    return should_indent
+
+
 def create_enum_wrapper(model_name, name, values):
-    enum_name = model_name + "_" + name
-    res = Template("""  enum class $enum_name { $enumerators, NUM_ITEMS };
-""").substitute(enum_name=enum_name, enumerators=", ".join(values))
+    enum_name = f"{model_name}_{name}"
     wrapper = name + "_wrapper"
-    res = res + Template("""  struct $wrapper : public json::jsonable  {
-        $wrapper() = default;
-        virtual std::string to_json() const {
-            switch(v) {
-        """).substitute({'wrapper' : wrapper})
-    for enum_entry in values:
-        res = res + "      case " + enum_name + "::" + enum_entry + ": return \"\\\"" + enum_entry + "\\\"\";\n"
-    res = res + Template("""      default: return \"\\\"Unknown\\\"\";
+
+    def indent(s, level):
+        return textwrap.indent(s, level * '    ', not_first())
+
+    case_clauses = "\n".join(
+        Template('''case $enum_name::$enum_entry: return "\\"$enum_entry\\"";''').substitute(
+            enum_name=enum_name, enum_entry=enum_entry) for enum_entry in values)
+    res = Template("""
```

Comment:
e.g. add a backslash here.

Reply:
okay, added. but as put above, this is the only case where we can benefit from the backslash.